dashboard/src/components/TestsTable/TestsTable.tsx

```diff
@@ -255,14 +258,17 @@ export function TestsTable({
     [filterCount, intl, filter],
   );
 
-  // TODO: there should be a filtering for the frontend before the backend AND that filtering should consider the individual tests inside each test "batch" (the data from individualTestsTables), not only the rows of the external table
+  // TODO: there should be a filtering for the frontend before the backend
+  // that frontend filtering should consider the individual tests inside each test batch (the data from individualTestsTables),
+  // not only the rows of the external table (same for BootsTable)
   const onSearchChange = useCallback(
     (e: React.ChangeEvent<HTMLInputElement>) => {
       if (e.target.value !== undefined && updatePathFilter) {
         updatePathFilter(e.target.value);
       }
-      // TODO: remove this frontend filter when the hardwareDetails backend filtering gets in place
-      table.setGlobalFilter(String(e.target.value));
+      if (updatePathFilter === undefined) {
+        table.setGlobalFilter(String(e.target.value));
+      }
```

Comment:
do we need to set the global filter? since the filter is being added in the backend

Reply:
In this case it is being used for tables such as within the buildDetails page, which doesn't have backend filtering for the page so the frontend filtering is being used